### ISSUE TITLE
Signage Accessibility Review

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#9156]
# Motivation

We want to make sure our Signage solution is meeting accessibility requirements and guidelines that have been shared with us.

# Additional Info

Some information https://github.com/orgs/uiowa/projects/20?pane=issue&itemId=96458842

Things to consider:

- font sizes
- color contrast
- links to alternate ways of accessing information in accessible format
- table captions
- more

# Scope

Scope to reviewing and identifying areas we should invest in. Scope does not include accessibility enhancements as part of this slice but we should have follow up issues created for anything identified before closing out this issue.

<details><summary>Tasklist</summary>
<p>

- [x] font sizes
- [x] color contrast
- [x] links to alternate ways of accessing information in accessible format
- [x] table captions
- [x] Making admin experience WCAG 2.1AA compliant
- [x] more

</p>
</details> 

# Findings
- We should design for 10 feet away.
    - This means our minimum font size should be 1 inch, and then after that we should scale it based on screen width so that it stays consistent on larger or more pixel dense screens. We can use something similar to  `font-size: max(1in, Xdvw);`
    - How can we do this without also making the admin experience bad? We don't want the admins to have to work with 1 inch text on their laptops.
    - [Here is a neat sizing tool I found for visualizing distant text.](https://www.uprinting.com/banner-font-sizes.html)
- For signs, we should implement an even stronger contrast ratio standard than normal. 
    - [The International Sign Association recommends a 65-70% contrast ratio for sign text and glyphs.](https://signs.org/codes-regulations/technical-codes-and-standards/ada-accessible-signage/sign-contrast/)
    - They recommend using the LRV value to calculate this. [Here is a calculator to find this ratio.](https://lrvcalculator.com/)
        - Sidenote, LRV means "light reflectance value", so its more meant for paint. We can sill use it, though it may be slightly incorrectly calculated depending on the reflectance of the screen it is displayed on.
- We should find some way to add QR codes to each slide type.
    - This could be a field on images, an automatic link to a filtered events listing, etc.
    - There are a few different ways to generate QR codes as modules ([Example 1](https://www.drupal.org/project/qrcode_generator), [Example 2](https://www.drupal.org/project/qrcode_fields), [Example 3](https://www.drupal.org/project/qr_generator)), but none supported by security advisory policy. Should we roll our own with [something like This](https://davidshimjs.github.io/qrcodejs/), or should we work with a module to create these?
- We should remove headings from table captions.
    - Originally crafted for the Bus widget, table captions currently hold some of our headings for slides. 
    - We should remove these and migrate the Headings to the slide paragraph type so that it is a universal field, and then re-style/re-template it to fit the correct heading structure and style of slides.
    - We should also be auto generating captions for tables, something like "A table of events something something", and then render them in an invisible caption.
 - Force captions to display on videos.
 - We should add back in the site chrome on the view page and then add controls (potentially both keyboard and buttons) to the slides.
 - We should have a sign preview button that removes all site chrome and shows it as if you are logged out.
     - Unfortunately I don't think there is a way to open an incognito browser with JS, so we will most likely have to emulate a logged out experience.
     - Humorous suggestion: Just add a logout button that we title "Sign preview" that sends them back to the sign after logout.
 - I did siteimprove scans on the pages of the admin experience for a user and found only a few dings, all of which are Drupal core specific.

**Comment by joewhitsitt** [comment]
noticed there was no h1 when viewing slides with other headings (h2, h3).

**Comment by GaryRidgway** [comment]
I recall there being some intent to sending out communications for this issue when we groomed it. Does anyone remember that as well? Its not in the notes, but I thought I would ask.

**Comment by GaryRidgway** [comment]
Something I want to remember for this is the phrase "Design for X feet away". The font sizing etc will become more concrete when we know how far away we want to design for.

**Comment by briand44** [comment]
We didn't plan on sending any comms related to this. This is just a review to identify areas we need to invest in. We discussed sending an announcement in https://github.com/uiowa/uiowa/issues/9175?

I think as part of this, we can reach out to TM if we have any questions around Signage accessibility but not sure of any other communications related to this.

**Comment by GaryRidgway** [comment]
[Here's a link to the original measuring I did.](https://github.com/uiowa/uiowa/issues/8528#issuecomment-3032857387)
[Here is a neat banner sizing tool](https://www.uprinting.com/banner-font-sizes.html)

**Comment by GaryRidgway** [comment]
Something I think we need to keep in mind for this is pixel DENSITY of a sign. Some signs might have a really dense pixel display, resulting in smaller text. Perhaps we should be sizing the font based on percentages of screen size?

**Comment by GaryRidgway** [comment]
Talked to Jake S today and he told me that generally his goto rule is to design for 10 ft away. Minimum font size should be roughly 1 inch for every 10 feet, so about 72pt font is the minimum we should have on our signs.

**Comment by GaryRidgway** [comment]
As I am thinking about how this might be implemented, I am caught by the issue of a styled `1inch` minimum font size on the editor side being very large. There has to be some way for us to implement a 1 inch minimum, but then allow for it to not look huge on a laptop while people are editing.

**Comment by GaryRidgway** [comment]
Just did some quick math to discern what size our current heading would be on our smallest sign, please point out any mathematical errors I have made

**My laptop width**
11.85in

**Width of our smallest sign**
34.9in

**Height of our current Heading on my laptop**
 0.14in

**Cross multiply**
```math
{11.85in\over 34.9in}= {0.138in\over X}
```
```math
4.82in= 11.85in * X
```
```
```math
0.41in = X
```

So currently our heading would be 0.41 inches on our smallest sign.

**Comment by GaryRidgway** [comment]
Notes from meeting:

- Did we capture the “force captions” video thing from demo? Is this the right place?
- Slideshow controls, table captions, alt text seem like the remaining items to consider.
- Scans with things. Show if things get wonky.

**Comment by briand44** [comment]
Going to close this as the review has finished. I will include the findings in separate issues as we transition to focus on ADA Title II compliance.
